p/scripts/main.js

```diff
@@ -1954,21 +1955,25 @@ function init_confirm_action() {
 	document.querySelectorAll('button.confirm').forEach(function (b) { b.disabled = false; });
 }
 
-function init_scroll_header() {
-	const header = document.querySelector('header.header');
-	if (header) {
-		const headerHeight = parseInt(getComputedStyle(header).height);
+function on_scroll_header() {
+	if (on_scroll_header.header === undefined) {
+		on_scroll_header.header = document.querySelector('header.header');
+	}
+	const header = on_scroll_header.header;
+	if (!header) {
+		return;
+	}
+	if (!header.headerHeight) {
+		header.headerHeight = parseInt(getComputedStyle(header).height);
 		header.previousScroll = 0;
-		window.addEventListener('scroll', function () {
-			const currentScroll = window.scrollY;
-			if (currentScroll > headerHeight && currentScroll > header.previousScroll) {
-				header.classList.add('hide');
-			} else {
-				header.classList.remove('hide');
-			}
-			header.previousScroll = currentScroll;
-		});
 	}
+	const currentScroll = window.scrollY;
+	if (currentScroll <= header.headerHeight || currentScroll < header.previousScroll - 4) {
+		header.classList.remove('hide');
+	} else if (currentScroll > header.previousScroll + 8) {
```

Comment:
I also strongly dislike this tbh. :-/

Reply:
It is not working so well anyway, but without it, there were far too many shifts between showing/hiding the bar